addons/product_visible_discount/product_visible_discount.py

```diff
@@ -7,12 +7,10 @@
 class product_pricelist(osv.osv):
     _inherit = 'product.pricelist'
 
-    _columns ={
-        'visible_discount': fields.boolean('Visible Discount'),
-    }
-    _defaults = {
-         'visible_discount': True,
+    _columns = {
+        'visible_discount': fields.selection([('with_discount', 'Apply discount in the price'), ('without_discount', 'Keep public price, but set a discount')], string="Visible Discount"),
```

Comment:
can you rename with 'discount_policy' ? and string should be like "Discount policy on sale order line"